galpy/orbit/Orbits.py

```diff
@@ -5597,28 +5652,566 @@ def animate(self,*args,**kwargs): #pragma: no cover
+        all_zlabel= kwargs.get('zlabel',[None for d in d3])
+        for d1,d2,d3, xlabel, ylabel, zlabel in zip(d1s,d2s,d3s,all_xlabel,all_ylabel,all_zlabel):
+            #Get x and y for each subplot
+            kwargs['dontreshape']= True
+            x= self._parse_plot_quantity(d1,**kwargs)
+            y= self._parse_plot_quantity(d2,**kwargs)
+            z= self._parse_plot_quantity(d3,**kwargs)
+            kwargs.pop('dontreshape')
+            xs.append(x)
+            ys.append(y)
+            zs.append(z)
+            if xlabel is None:
+                xlabels.append(labeldict.get(d1,r'\mathrm{No\ xlabel\ specified}'))
+            else:
+                xlabels.append(xlabel)
+            if ylabel is None:
+                ylabels.append(labeldict.get(d2,r'\mathrm{No\ ylabel\ specified}'))
+            else:
+                ylabels.append(ylabel)
+            if zlabel is None:
+                zlabels.append(labeldict.get(d3,r'\mathrm{No\ ylabel\ specified}'))
+            else:
+                zlabels.append(zlabel)
+        kwargs.pop('ro',None)
+        kwargs.pop('vo',None)
+        kwargs.pop('obs',None)
+        kwargs.pop('use_physical',None)
+        kwargs.pop('pot',None)
+        kwargs.pop('OmegaP',None)
+        kwargs.pop('quantity',None)
+        width= kwargs.pop('width',800)
+        height= kwargs.pop('height',600)
+        load_jslibs= kwargs.pop('load_jslibs',True)
+        if load_jslibs:
+            load_jslibs_code= """
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js"></script>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.1/jquery.min.js"></script>
+    """
+        else:
+            load_jslibs_code= ""
+        # Dump data to HTML
+        nplots= len(xs)
+        jsonDict= {}
+        for ii in range(nplots):
+            for jj in range(self.size):
+                jsonDict['x%i_%i' % (ii+1,jj)]= xs[ii][jj].tolist()
+                jsonDict['y%i_%i' % (ii+1,jj)]= ys[ii][jj].tolist()
+                jsonDict['z%i_%i' % (ii+1,jj)]= zs[ii][jj].tolist()
+        jsonDict["time"] = self.time().tolist()
+        json_filename= kwargs.pop('json_filename',None)
+        if json_filename is None:
+            jd= json.dumps(jsonDict)
+            json_code= f"""  let data= JSON.parse('{jd}');"""
+            close_json_code= ""
+        else:
+            with open(json_filename,'w') as jfile:
+                json.dump(jsonDict,jfile)
+            json_code= f"""Plotly.d3.json('{json_filename}',function(data){{"""
+            close_json_code= "});"
+        self.divid= 'galpy-'\
+            +''.join(choice(ascii_lowercase) for i in range(24))
+        button_width= 419.51+4.*10.
+        button_margin_left= int(numpy.round((width-button_width)/2.))
+        if button_margin_left < 0: button_margin_left= 0
+        # Configuration options
+        config= """{{staticPlot: {staticPlot}}}"""\
```

Comment:
Does `staticPlot` actually work for 3D? Doesn't seem to work for me in VS Code.

Reply:
I have removed `staticPlot` for animate3d. Seems like a plotly.js issue (https://github.com/plotly/plotly.js/issues/457)